src/nodejs/hooks/seeker_agent_test.go

```diff
@@ -0,0 +1,282 @@
+				entryPointPath := filepath.Join(buildDir, "server.js")
+				Expect(entryPointPath).ToNot(BeAnExistingFile())
+				const mockedCode = "some mock javascript code"
+				Expect(ioutil.WriteFile(entryPointPath, []byte(mockedCode), 0755)).To(Succeed())
+				err = seeker.AfterCompile(stager)
+				Expect(err).To(BeNil())
+				contents, err := ioutil.ReadFile(entryPointPath)
+				Expect(err).To(BeNil())
+				Expect(string(contents)).To(Equal(hooks.SeekerRequire + mockedCode))
+			})
+			It("does not prepend 'require('./seeker/node_modules/@synopsys-sig/seeker);' to the server.js file if it already exist", func() {
+				entryPointPath := filepath.Join(buildDir, "server.js")
+				Expect(entryPointPath).ToNot(BeAnExistingFile())
+				const mockedCode = hooks.SeekerRequire + "some mock javascript code"
+				Expect(ioutil.WriteFile(entryPointPath, []byte(mockedCode), 0755)).To(Succeed())
+				err = seeker.AfterCompile(stager)
+				Expect(err).To(BeNil())
+				contents, err := ioutil.ReadFile(entryPointPath)
+				Expect(err).To(BeNil())
+				Expect(string(contents)).To(Equal(mockedCode))
+			})
+			It("Fails to prepend the require to the server.js file - when the file does not exist", func() {
+				entryPointPath := filepath.Join(buildDir, "server.js")
+				Expect(entryPointPath).ToNot(BeAnExistingFile())
+				err = seeker.AfterCompile(stager)
+				Expect(err).NotTo(BeNil())
+				Expect(err.Error()).To(ContainSubstring("no such file or directory"))
+			})
+
+		})
+	})
+})
+
+func getMockedAgentDownloader() MockDownloader {
+	return MockDownloader{Mock: func(url, path string) error {
+		const expectedUrlSuffix = "rest/api/latest/installers/agents/binaries/NODEJS"
+		if !strings.HasSuffix(url, expectedUrlSuffix) {
+			return errors.New("expected to be called with url that ends with " + expectedUrlSuffix)
+		}
+		return nil
+	}}
+}
+
+func getAgentUnzipper() MockUnzipper {
+	return MockUnzipper{Mock: func(zipFile, absoluteFolderPath string) error {
+		s, e := getFixtureAbsolutePath("NODEJS_agent.zip")
+		if e != nil {
+			return e
+		}
+		z := hooks.SeekerUnzipper{}
+		return z.Unzip(s, absoluteFolderPath)
+	}}
+}
+
+func getFixtureAbsolutePath(fileName string) (string, error) {
+	path, err := filepath.Abs("../../../fixtures/seeker/" + fileName)
+	return path, err
+}
+
+type MockDownloader struct {
+	Mock func(url, path string) error
+}
+
+func (f MockDownloader) DownloadFile(url, path string) error {
+	return f.Mock(url, path)
+}
+
+type MockUnzipper struct {
+	Mock func(zipFile, absoluteFolderPath string) error
+}
+
+func (u MockUnzipper) Unzip(zipFile, absoluteFolderPath string) error {
+	return u.Mock(zipFile, absoluteFolderPath)
+}
+
+type MockSeekerCommand struct {
+}
+
+func (s MockSeekerCommand) Execute(dir string, stdout io.Writer, stderr io.Writer, program string, args ...string) error {
+	Expect(program).To(Equal("npm"))
+	Expect(args).To(Equal([]string{"install", "--save", "/tmp/seeker-agent.tgz", "--prefix", "seeker"}))
```

Comment:
This link is causing unit tests to fail. `/tmp/`does not work well across all systems.
Please change `/tmp/seeker-agent.tgz` => `filepath.Join(os.TempDir(), "seeker-agent.tgz")`

Reply:
fixed